### ISSUE TITLE
Move AR.js repositories to a github organisation

### ISSUE THREAD

**Issue by jeromeetienne** [jeromeetienne/AR.js#253]
- suggested by @ZoltanVeres
- created the organisation - https://github.com/ARjs-org
- transfer ownership of AR.js repository there https://github.com/jeromeetienne/AR.js
- transfer ownership of ARjs-docs too https://github.com/jeromeetienne/AR.js-docs

**Comment by jeromeetienne** [comment]
i never transfered the ownership of a repo... is there a risk ? i dont wanna break AR.js repo :)

**Comment by jeromeetienne** [comment]
## Github Pages are not forwarded
Based on github docs, github pages links are not forwarded, How to handle this ?

![screen shot 2017-11-20 at 13 23 00](https://user-images.githubusercontent.com/252962/33020422-2644b082-cdf6-11e7-9467-dd0fea133c9e.png)

**Comment by ZoltanVeres** [comment]
Hi,

I don't know to be honest, we should definitely investigate it!

**Comment by ZoltanVeres** [comment]
We should make a list of what GH pages are in the repository and move them
to the right place first (ex. to AR-docs, or somewhere else).
Ideally i would prefer to not have GH pages in AR.js repository, only html
pages for testing purposes (for now)
And all pages/demos should be moved to AR-docs, pointing to a domain.
If this is done, then we can migrate the repos, and providing a new url to
the examples and external links.

**Comment by raisedadead** [comment]
@jeromeetienne 

I have done so before.

TL;DR:
relative links will work, hard links might break

> - What happen to the previous links ?
> - does github handle all the http forwarding ?
> - do we need to do that ourselves ?

`https://github.com/jeromeetienne/AR.js` -> `https://github.com/ARjs-org/AR.js`

This redirection will persist automatically howsoever times you change the repo name / org name. Happy renaming.

But that said your will have to consider the links that are hard-linked in your documentations, those are the only things that will break.

That said how about a more catchy name like **getarjs** for the org?
https://github.com/getarjs

I have the ownership to this. I can transfer to you should you want it?

**Comment by raisedadead** [comment]
Also, if you are moving stuff you should do so as early stage as possible. The breaks that will happen is the anybody doing a local clone of the your repo directly will have to update their remotes I guess.

Forks will be fine though.

**Comment by raisedadead** [comment]
For GitHub Pages you will have not much choice with the standard `*.github.io` domains. But to future proof the links you should:

Check out https://github.com/js-org/dns.js.org and get a subdomain that suites you.
like `http://ar.js.org`. They are awesome!

A CNAME file in the `gh-pages` directory will auto redirect to a domain like `ar.js.org`

Now coming back to current links like `https://jeromeetienne.github.io/AR.js` which have been linked from elsewhere on the internet, they will break after the transfer unfortunately.

**Comment by raisedadead** [comment]
And I did some quick searches that confirmed my doubts. The only hack right now seems to be this workaround

https://gist.github.com/domenic/1f286d415559b56d725bee51a62c24a7

**Comment by raisedadead** [comment]
@jeromeetienne @ZoltanVeres Hi again.

Any chance that you guys are still interested in the @getarjs handle for the org name? I can hand it off to you if you are still interested.

Cheers

**Comment by ZoltanVeres** [comment]
@raisedadead I don't think so. @jeromeetienne bought a domain for AR.js and we already created the GH organisation [https://github.com/orgs/ARjs-org](ARjs-org). We are in the process of moving stuff there.

**Comment by janpio** [comment]
As I created #469, I also restarted an investigation on how to move the repo to the org.

The Github Pages problem seems to still exist:
https://github.com/isaacs/github/issues/492

I tweeted at Github support to ask, if there is newer information on this:
https://twitter.com/Sujan/status/1094925752028577793

---

With _some_ additional work, and some disadvantages in another area, one could work around it. From the support response at https://github.com/isaacs/github/issues/492:

> In the meantime, you could create a new repository in place of the transferred one that links to the repository's new home. This would eliminate the redirect that we do have on the repository page, but its up to you and your collaborators to decide if removing the automatic redirect on GitHub.com is worth placing manual redirects on GitHub Pages and GitHub.com

So after transferring, one could create `jeromeetienne/AR.js` again and thus breaking the automatic redirect to the new repo. But then one can place a README with a link to the new repo, which is a slight disadvantage but still ok. 
Then one has to recreate all the page that were hosted on GitHub Pages and put a redirect to the new domain in there ([This blog post](https://blog.kevin-brown.com/github/2015/01/15/transferring-repositories-cleanly-on-github.html) about this topic contains some example code, but we should probably use something more elaborate to redirect to the absolut correct pages instead of just the root). This would cause all the _old_ links to still work and automatically redirect to the new URLs of the new org repo.

(Of course fully depends on the original repo owner's cooperation and help)

But if "keep the old URLs working" is a hard requirement, and as those are linked in lots of blog posts and other places I think it should be, this would be one option.

---

Some context:
Those are all the URLs we are talking about here: https://www.google.com/search?q=site%3Ajeromeetienne.github.io (~400 results right now)
(Let's hope GitHub support has another, simpler one.)

**Comment by nicolocarpignoli** [comment]
closed, see related PRs.
